### ISSUE TITLE
Linaria crashes with `EvalError: Unexpected token 'export'`

### ISSUE THREAD

**Issue by nstepien** [callstack/linaria#1242]
## Environment

- Linaria version: `@linaria/webpack5-loader@4.1.16`
- Bundler (+ version): `webpack@5.82.0`
- Node.js version: 20.0.0
- OS: win11

## Description

Linaria crashes with this error with the minimal reproducible demo below:
```
> webpack

assets by status 1.85 KiB [cached] 1 asset
./main.js 39 bytes [built] [code generated] [1 error]

ERROR in ./main.js
Module build failed (from ./node_modules/@linaria/webpack5-loader/lib/index.js):
D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\module.js:355
      throw new EvalError(`${e.message} in${callstack.join('\n| ')}\n`);
      ^

EvalError: Unexpected token 'export' in
| D:\repos\vite-bug\node_modules\clsx\dist\clsx.m.js
| D:\repos\vite-bug\node_modules\react-data-grid\lib\bundle.js
| D:\repos\vite-bug\main.js

    at Module.evaluate (D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\module.js:355:13)
    at require.Object.assign.ensure (D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\module.js:295:13)       
    at D:\repos\vite-bug\node_modules\react-data-grid\lib\bundle.js:21:36
    at D:\repos\vite-bug\node_modules\react-data-grid\lib\bundle.js:339:3
    at Script.runInContext (node:vm:134:12)
    at Module.evaluate (D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\module.js:342:14)
    at require.Object.assign.ensure (D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\module.js:295:13)       
    at D:\repos\vite-bug\main.js:3:45
    at D:\repos\vite-bug\main.js:7:3
    at Script.runInContext (node:vm:134:12)
    at Module.evaluate (D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\module.js:342:14)
    at evaluate (D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\evaluators\index.js:20:5)
    at evalStage (D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\transform-stages\2-eval.js:26:45)
    at syncStages (D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\transform.js:47:45)
    at transform (D:\repos\vite-bug\node_modules\@linaria\babel-preset\lib\transform.js:146:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

webpack 5.82.0 compiled with 1 error in 865 ms
```

## Reproducible Demo

https://github.com/nstepien/vite-bug/tree/webpack2

1. Clone the repo above
1. checkout the `webpack2` branch
1. `npm ci`
1. `npm run build`

What's really odd is that if I update `react-data-grid` to `7.0.0-beta.29` in this repo, the error goes away and webpack builds fine.
The `export` error suggest it comes from `clsx`, but `clsx` doesn't change.
Only the linaria babel preset has issues with `react-data-grid\lib\bundle.js`, so I don't think there's anything wrong with the `react-data-grid` package itself.

**Comment by rolftimmermans** [comment]
I ran into this as well. I have a fix here: https://github.com/callstack/linaria/compare/master...rolftimmermans:linaria:master

If you think this is the right approach I'd be happy to open a PR.

**Comment by Anber** [comment]
Hi @nstepien!

Thank you for the report! It appears that the rule in charge of identifying ESM files contained an incomplete regular expression. The issue has been resolved in #1245.

**Comment by nstepien** [comment]
Thank you!
